tests/app/main/test_frameworks.py

```diff
@@ -5751,3 +5750,136 @@ def test_false_values(self, falsey_value):
+        assert res.status_code == 200
+        assert "You cannot sign the Framework Agreement" in res.get_data(as_text=True)
+
+
+class TestSignFrameworkAgreement(BaseApplicationTest):
+    """Tests for app.main.views.frameworks.sign_framework_agreement"""
+
+    def setup_method(self, method):
+        super().setup_method(method)
+        self.data_api_client_patch = mock.patch('app.main.views.frameworks.data_api_client', autospec=True)
+        self.data_api_client = self.data_api_client_patch.start()
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super().teardown_method(method)
+
+    @pytest.mark.parametrize(
+        ('framework_slug', 'on_framework', 'status_code'),
+        (
+            ('g-cloud-11', True, 404),
+            ('g-cloud-12', True, 200),
+            ('g-cloud-12', False, 400),
+        )
+    )
+    def test_only_works_for_supported_frameworks(self, framework_slug, on_framework, status_code):
+        self.login()
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug=framework_slug,
+                                                                         framework_agreement_version="1")
+        self.data_api_client.find_draft_services_iter.return_value = [
+            {'serviceName': 'A service', 'status': 'submitted', 'lotSlug': 'iaas'}
+        ]
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=on_framework)
+
+        res = self.client.get(f"/suppliers/frameworks/{framework_slug}/sign-framework-agreement")
+        assert res.status_code == status_code
+
+    def test_post_signs_agreement(self):
+        self.login()
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug='g-cloud-12',
+                                                                         framework_agreement_version="1")
+        self.data_api_client.find_draft_services_iter.return_value = [
+            {'serviceName': 'A service', 'status': 'submitted', 'lotSlug': 'iaas'}
+        ]
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=True)
+
+        res = self.client.get(f"/suppliers/frameworks/g-cloud-12/sign-framework-agreement")
+        assert res.status_code == 200
+        res = self.client.post(f"/suppliers/frameworks/g-cloud-12/sign-framework-agreement",
+                               data={"signerName": "Jane Doe",
+                                     "signerRole": "Director",
+                                     "signer_terms_and_conditions": "True"})
+        self.data_api_client.create_framework_agreement.return_value = {"agreement": {"id": 789}}
```

Comment:
Should this be set before the client calls?

Reply:
Yes, looks like that was the issue 😅